### ISSUE TITLE
I want to adapt your algorithm to other types object counting.

### ISSUE THREAD

**Issue by 13052** [diptodip/counting#5]
First I want to thank you for your paper and this repository, which really helped me a lot. Now I am working on a counting task but not for counting people, also each image could have several hundreds objects. This may different to human crowd counting. 

The annotation point, should it be the center of an object or other part, because these objects not like humans always stand on the ground and up head. For instance I want to count how many animals in an image, animals could have a lot more complicated poses and occlusions. 

I am new to computer vision area, some detailed information within your paper and reference papers currently not very clear to me, so I want to know what directions should I go? What hyper parameters should I adjust for this task?

I have already trained a AMDCN model about animal counting, it performs good, but due to lack of image types, the generalization ability of the model is bad. Anyway, I tried to change some hyper parameters, then I found out that none of these models performed better than your original default hyper parameters. Why did this happen? the task is very different. So I'm here to get some inspiration from you.

Thank you!

**Comment by diptodip** [comment]
Hi, thanks for the kind words! I'm also new to computer vision so I'm glad this was helpful.

I'll try to address your questions in order:

1. You're right, if you have animals with varied poses then making the annotations becomes trickier. I would still put the dots on the head, because as long as the animals are facing somewhat towards the camera, you can take advantage of the distinct structure of faces. Actually, most of the datasets we used did the same thing even on humans because it's too tricky to annotate bodies. There was one annotation style that separated bodies and heads.

2. Some of the hyperparameters here are:
    i. the blurring amount (Σ) when creating the sum of Gaussians density map.
    ii. the network architecture &mdash; there are a lot of things you could play with here like number of layers, kernel sizes on each layer, dilations on each layer; you could even add more modules like a [U-net](https://arxiv.org/abs/1505.04597) either to the beginning or end of the network (not sure which would help better if it would help at all).
    iii. the annotation model &mdash; if you know the kind of animal has a particular body shape in the datasets you're using, you could construct a separate head and body annotation that, when blurred, make a single shape. This is something that was done in [27](http://www.ee.cuhk.edu.hk/~xgwang/papers/zhangLWYcvpr15.pdf), though it actually turned out to not be helpful for the network in my paper. You could also predict counts for different types of animals separately; either by using a different model for each kind of animal or by having one network do a kind of multitask learning by outputting a multi-channel density map with each channel containing the density map for a different type of animal. Of course, you then have to make your ground truth density map follow the same format.
Not sure exactly what kinds of details you found confusing, but hopefully these are helpful.

3. Once you've decided which hyperparameters you want to change, there's no easy way to choose the appropriate values. What you really want to do is a grid search across *all combinations* of your hyperparameters with cross validation on a validation set of your data. The hyper parameters used here were chosen that way for these datasets, but they all work together. For example, if you change Σ when generating your labels, you may also have to change the number of layers or kernel sizes in the network architecture. If you're generating the labels in exactly the same way then I think even if it's a different kind of data it might be hard to improve the network hyperparameters.

**Comment by 13052** [comment]
Thank you for your so detailed reply, this is really helpful, I will try your suggestions. Thanks again!!!
